### PR TITLE
 bootloader: use ForGadget when installing boot config 

### PR DESCRIPTION
**Pull request by bboozzoo**
Address TODO added during UC20 development and use a ForGadget helper when
installing boot configs of a bootloader.

### CODE REVIEW

**Review by anonymouse64 — APPROVED** [review]
LGTM, thanks for this

File: bootloader/bootloader.go [hunk 1]
```diff
@@ -249,16 +246,11 @@ func InstallBootConfig(gadgetDir, rootDir string, opts *Options) error {
...
-	// TODO:UC20 use ForGadget() to obtain the right bootloader
-	for _, bl := range []installableBootloader{&grub{}, &uboot{}, &androidboot{}, &lk{}} {
-		bl.setRootDir(rootDir)
-		ok, err := bl.InstallBootConfig(gadgetDir, opts)
-		if ok {
-			return err
-		}
+	bl, err := ForGadget(gadgetDir, rootDir, opts)
+	if err != nil {
+		return fmt.Errorf("cannot find boot config in %q", gadgetDir)
```
[the remark this answers is not in the document]
Comment (mvo5) [reply bootloader/bootloader.go:250]: Yeah, this is nice!

**Review by mvo5 — APPROVED** [review]
Thank you

**Review by pedronis — APPROVED** [review]
thank you

**Pull request merged**